forum/managers.py

```diff
@@ -0,0 +1,39 @@
+from rest_framework_simplejwt.exceptions import (AuthenticationFailed,
+                                                 TokenError)
+from rest_framework_simplejwt.tokens import AccessToken, RefreshToken
+
+from .errors import Error
+
+
+class TokenManager:
+
+    @classmethod
+    def generate_token_for_user(cls, user) -> str:
+        """Generates token for user"""
+
+        access_token = AccessToken.for_user(user)
+        return str(access_token)
+
+    @classmethod
+    def generate_refresh_token_for_user(cls, user) -> str:
+        """Generates token for user"""
+
+        refresh_token = RefreshToken.for_user(user)
+        return str(refresh_token)
+
+    @classmethod
+    def get_payload(cls, token) -> dict:
+        """Returns payload for token"""
+
+        decoded_token = cls.__get_decoded_token(token)
```

Comment:
This method assumes that every token passed to it will be a valid `AccessToken`. Consider cases where you might need to decode different types of tokens (e.g., `RefreshToken`). You might either handle different token types in this method or create separate methods for different token types if their handling diverges significantly

Reply:
I choose the second method to create separate methods for different token types, to avoid nested try-except blocks.